spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -270,8 +270,17 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
   public void invalidateTable(Identifier ident) {
     try {
       load(ident).first().refresh();
-    } catch (org.apache.iceberg.exceptions.NoSuchTableException ignored) {
-      // ignore if the table doesn't exist, it is not cached
+    } catch (Exception e) {
```

Comment:
What are you trying to catch here? Are you attempting to catch the exception for the wrong table UUID?

Reply:
I have add `TableUUIDMismatchException` class.